### DISCUSSION TITLE
It supports a command like `pnpm run preview --host` when the output is the `server`

### DISCUSSION THREAD

**Discussion by JerryWu1234** [withastro/roadmap#493] [Proposal]
### Body

# Summary
Add a `--host` flag to `astro preview` when the output is `server`
![image](https://user-images.githubusercontent.com/27261522/227081885-c6caddbb-325e-410e-8561-41b61464dd96.png)
# Background & Motivation

I'm a mobile developer used to developing with the h5 for reduced size.
Recently I used Astro to develop an h5 application and preview it before deploying.
It can preview when the output of the config is static. 
But when the output of config is server not well.

# Goals

I can use `pnpm run preivew  --host`

# Example

`astro preview --host`

**Comment by matthewp** [comment]
Hey @wulinsheng123 thanks for filing this. What is it that `--host` does? We already do support `astro preview` with server output. However not all adapters support it; the adapter must. For example `@astrojs/node` does support it. Does the `--host` flag do something extra?

**Reply by jasikpark** [reply]
For a vanilla Vite server `--host` binds to `0.0.0.0` instead of localhost for access on mobile devices, for example. 

https://vitejs.dev/config/server-options.html#server-host

**Reply by JerryWu1234** [reply]
![image](https://user-images.githubusercontent.com/27261522/227081620-bcd1d793-42f5-4d58-a2f8-972180b8df71.png)
I want to access it by the Network address.

**Comment by florian-lefebvre** [comment]
You can use `--host` now: https://docs.astro.build/en/reference/cli-reference/#--host-optional-host-address
